internal/resolution/variable_sources/global_constraints/global_constraints.go

```diff
@@ -0,0 +1,101 @@
+package global_constraints
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/operator-framework/deppy/pkg/deppy"
+	"github.com/operator-framework/deppy/pkg/deppy/constraint"
+	"github.com/operator-framework/deppy/pkg/deppy/input"
+
+	"github.com/operator-framework/operator-controller/internal/resolution/variable_sources/bundles_and_dependencies"
+	olmentity "github.com/operator-framework/operator-controller/internal/resolution/variable_sources/entity"
+)
+
+type GlobalConstraintsVariable struct {
+	*input.SimpleVariable
+}
+
+func NewGlobalConstraintsVariable(id deppy.Identifier, atMostIDs ...deppy.Identifier) *GlobalConstraintsVariable {
+	return &GlobalConstraintsVariable{
+		SimpleVariable: input.NewSimpleVariable(id, constraint.AtMost(1, atMostIDs...)),
+	}
+}
```

Comment:
I'm not sure that I would infer that the`NewGlobalConstraintsVariable` function name implies that a single entity will be selected. Could you explain why it can be infered or add a comment or change the name?

Reply:
Yeah, that's fair. Let me see if I can find a better name. I think I've been using global constraints for so long I just assumed it was a thing =P

Reply:
I've tried to refactor to a different name - hopefully it will be more descriptive